datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -540,6 +541,8 @@ async fn fetch_statistics(
 pub struct ParquetSink {
     /// Config options for writing data
     config: FileSinkConfig,
+    /// File metadata from successfully produced parquet files.
```

Comment:
To use this API in a more general context, we would probably need to define how these `FileMetaData` mapped to the original files

Maybe it could be something like this (which would also support the API below)


```rust
HashMap<Path, FileMetaData>
```